server/src/main/java/org/apache/gravitino/server/web/rest/MetadataObjectPolicyOperations.java

```diff
@@ -0,0 +1,357 @@
+                  policyNames.length,
+                  type,
+                  fullName,
+                  metalake);
+              return Utils.ok(new NameListResponse(policyNames));
+            }
+          });
+
+    } catch (Exception e) {
+      return ExceptionHandlers.handlePolicyException(OperationType.LIST, "", fullName, e);
+    }
+  }
+
+  @POST
+  @Produces("application/vnd.gravitino.v1+json")
+  @Timed(name = "associate-object-policies." + MetricNames.HTTP_PROCESS_DURATION, absolute = true)
+  @ResponseMetered(name = "associate-object-policies", absolute = true)
+  public Response associatePoliciesForObject(
+      @PathParam("metalake") String metalake,
+      @PathParam("type") String type,
+      @PathParam("fullName") String fullName,
+      PoliciesAssociateRequest request) {
+    LOG.info(
+        "Received associate policies request for object type: {}, full name: {} under metalake: {}",
+        type,
+        fullName,
+        metalake);
+
+    try {
+      return Utils.doAs(
+          httpRequest,
+          () -> {
+            request.validate();
+            MetadataObject object =
+                MetadataObjects.parse(
+                    fullName, MetadataObject.Type.valueOf(type.toUpperCase(Locale.ROOT)));
+            String[] policyNames =
+                policyDispatcher.associatePoliciesForMetadataObject(
+                    metalake, object, request.getPoliciesToAdd(), request.getPoliciesToRemove());
+            policyNames = policyNames == null ? new String[0] : policyNames;
+
+            LOG.info(
+                "Associated policies: {} for object type: {}, full name: {} under metalake: {}",
+                Arrays.toString(policyNames),
+                type,
+                fullName,
+                metalake);
+            return Utils.ok(new NameListResponse(policyNames));
+          });
+
+    } catch (Exception e) {
+      return ExceptionHandlers.handlePolicyException(OperationType.ASSOCIATE, "", fullName, e);
+    }
+  }
+
+  private boolean hasConflictExclusivePolicy(
+      String metalake, MetadataObject object, Policy targetPolicy) {
+    if (!targetPolicy.exclusive()) {
+      return false;
+    }
+
+    Policy[] policies = policyDispatcher.listPolicyInfosForMetadataObject(metalake, object);
+    return Arrays.stream(policies)
+        .anyMatch(
+            p ->
+                p.exclusive()
+                    && p.policyType().equalsIgnoreCase(targetPolicy.policyType())
+                    && !p.name().equals(targetPolicy.name()));
+  }
+
+  private boolean hasConflictExclusivePolicy(Policy[] policies, Policy targetPolicy) {
+    if (!targetPolicy.exclusive() || ArrayUtils.isEmpty(policies)) {
+      return false;
+    }
+    return Arrays.stream(policies)
+        .anyMatch(
+            p ->
+                p.exclusive()
+                    && p.policyType().equalsIgnoreCase(targetPolicy.policyType())
+                    && !p.name().equals(targetPolicy.name()));
+  }
```

Comment:
What's the difference of these two methods? AFAIK, we can move the difference out of this method, so they can be combined as one.

What's more, the logic here in this file to check the different status is too complicated, you should add more comments. Which also makes me think that this policy system is a bit hard to understand and explain, because there're too many properties.

Reply:
fixed and added more comments